d2/src/main/java/com/linkedin/d2/balancer/util/ClusterInfoProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+   Copyright (c) 2019 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.d2.balancer.util;
+
+import com.linkedin.d2.balancer.ServiceUnavailableException;
+
+
+/**
+ * ClusterInfoProvider provides a mechanism to access detailed cluster information from the D2 infrastructure.
+ *
+ * @author David Hoa
+ * @version $Revision: $
+ */
+public interface ClusterInfoProvider
+{
+  String HTTPS_SCHEME = "https";
+  int DEFAULT_PARTITION = 0;
```

Comment:
Are there public static variables for these already defined somewhere that we can use?

Reply:
I switched to using DefaultPartitionAccessor.DEFAULT_PARTITION_ID, thanks for the suggestion. However, I did not find one for HTTPS_SCHEME. I added it to PropertyKeys, didn't find a better place for it.